pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -181,38 +190,69 @@ private Representation getSegmentFile(
   private Representation getSegmentsForTable(
       @Parameter(name = "tableName", in = "path", description = "The name of the table for which to list segments", required = true)
       String tableName,
-      @Parameter(name = "active", in = "query", description = "true = show active segments (in Helix), false = all segments (on filesystem)", required = false)
-      boolean activeOnly
-      ) {
+      @Parameter(name = "tableType", in = "query", description = "Type of table {offline|realtime}", required = false)
+      String type
+      ) throws Exception {
     Representation presentation;
+    JSONArray ret = new JSONArray();
+    final String realtime = "REALTIME";
+    final String offline = "OFFLINE";
+
+    if (type == null) {
+      ret.put(formatSegments(tableName, offline));
+      ret.put(formatSegments(tableName, realtime));
+    } else if (type.equalsIgnoreCase(realtime)) {
+      ret.put(formatSegments(tableName, realtime));
+    } else {
+      ret.put(formatSegments(tableName, offline));
+    }
+
+    presentation = new StringRepresentation(ret.toString());
+    return presentation;
+  }
+
+  private org.json.JSONObject formatSegments(String tableName, String tableType) throws Exception {
+    String realtimeTableName = TableNameBuilder.REALTIME_TABLE_NAME_BUILDER.forTable(tableName);
+    String offlineTableName = TableNameBuilder.OFFLINE_TABLE_NAME_BUILDER.forTable(tableName);
+
+    String tableNameWithType;
+    if (tableType.equalsIgnoreCase("REALTIME")) {
+      tableNameWithType = realtimeTableName;
+    } else {
+      tableNameWithType = offlineTableName;
+    }
+
+    org.json.JSONObject obj = new org.json.JSONObject();
+    obj.put(TABLE_NAME, tableNameWithType);
```

Comment:
You need only table type here. Not tableNameWithType. For an example see PinotTableRestletResource.getTable(). We are looking for something like this:
{ "REALTIME": ["rtseg1", "rtseg2"], "OFFLINE":["ofseg1", "ofseg2", "ofseg3"]}